### ISSUE TITLE
A11y: Allow customising parent element for screen reader live region to allow usage within `aria-modal` elements

### ISSUE THREAD

**Issue by diondiondion** [atlassian/react-beautiful-dnd#2332]
### Description

In our product, we're providing reordering functionality inside of a modal dialog.

As our modal dialog is marked up with the `aria-modal="true"` attribute, screen readers supporting this attribute will ignore all changes on the page outside of the modal.

Unfortunately, the live region used for announcing screen reader messages is always rendered directly into the body of the page, which means that the changes are not announced to our users.

It would be great if the element that the live region is rendered into could be customised so that this issue can be prevented.

**Comment by danielpickett** [comment]
@diondiondion Did you find a workaround for this? I'm dealing with this same issue right now. I'm using [react-beautiful-dnd](https://www.npmjs.com/package/react-beautiful-dnd) with [@reach/dialog](https://www.npmjs.com/package/@reach/dialog) and the live region is not announcing.

**Comment by diondiondion** [comment]
> Did you find a workaround for this?

Not a good one. I was able to "fix" it by simply removing the `aria-modal` attribute from our modal component. Since the attribute isn't super well-supported by screen readers, we polyfill what it does (hiding background content) anyway.

If you rely on the attribute, the only workaround I can think of would be basically rebuilding the live region yourself (if the library exposes the necessary hooks).

But at that stage, creating a fork that allows customising the parent element for the live region may be easier.

**Comment by danielpickett** [comment]
Thanks for the quick response @diondiondion. I'll have to look into the accessibility considerations of removing `aria-modal`. But then I probably couldn't use @reach/dialog 🤦‍♂️ . Maybe forking it is the best solution. Sigh.
